File: rosters.rb
#!/usr/bin/env ruby
#
location = File.dirname __FILE__

$: << "#{location}"
require 'ProgRunner'

class Conference
  attr_reader :name, :schools

  def initialize( name, schools )
    @name = name
    @schools = schools
  end

end


record_prog = ProgRunner.new location, "get_record"
stats_prog = ProgRunner.new location, "get_player_stats"

conferences = [ Conference.new( 'New England', [ 'uct', 'ume', 'uma', 'unh', 'uri', 'uvt' ] ),
                Conference.new( 'Atlantic',    [ 'ude', 'umd', 'unj', 'uny', 'uva', 'uwv' ] ),
                Conference.new( 'Southeast',   [ 'ual', 'ufl', 'uga', 'unc', 'usc', 'utn' ] ),
                Conference.new( 'Great Lake',  [ 'uil', 'uin', 'uky', 'umi', 'uoh', 'upa' ] ),
                Conference.new( 'Southwest',   [ 'uaz', 'uca', 'uco', 'unv', 'unm', 'uut' ] ),
                Conference.new( 'Northwest',   [ 'uid', 'umt', 'une', 'uor', 'uwa', 'uwy' ] ),
                Conference.new( 'Midwest',     [ 'uia', 'uks', 'umn', 'und', 'usd', 'uwi' ] ),
                Conference.new( 'South',       [ 'uar', 'ula', 'ums', 'umo', 'uok', 'utx' ] ) ]

Blue_Conferences = ['New England', 'Atlantic', 'Southeast', 'Great Lake']
Red_Conferences  = ['Southwest', 'Northwest', 'Midwest', 'South']

if ARGV.length > 0
  desired_conference = ARGV[0]
end

conferences.each do |conference|
  if !desired_conference.nil?
    if    desired_conference == "Blue"
      next if !Blue_Conferences.include? conference.name
    elsif desired_conference == "Red"
      next if !Red_Conferences.include? conference.name
    elsif desired_conference != conference.name
      next
    end
  end

  print "#{conference.name}\n"
  print "\n"

  conference.schools.each do |school|
    args = "#{school}.stat"

    record_prog.execute args

    (name, wins, losses, ties) = record_prog.get_output.split /,/

    printf "%s  %2d - %2d - %2d\n", name, wins, losses, ties
    printf "\n"

    stats_prog.execute args

    print stats_prog.get_output
    printf "\n"
  end
end

File: rosters.pl
#!/usr/bin/perl
#
# Reads filled in schedule.csv file and calculates
# Overall, Division and Head-To-Head records as well
# as scoring totals.
use File::Basename;

my ($location) = (fileparse( $0 ))[1];

my ($record_program) = $location . "get_record";
my ($stats_program)  = $location . "get_player_stats";

my (@conferences) = ( 'New England', 'Atlantic', 'Southeast', 'Great Lake', 'Southwest', 'Northwest', 'Midwest', 'South' );



my (%teams) = ( 'New England' => [ 'uct', 'ume', 'uma', 'unh', 'uri', 'uvt' ],
		'Atlantic'    => [ 'ude', 'umd', 'unj', 'uny', 'uva', 'uwv' ],
		'Southeast'   => [ 'ual', 'ufl', 'uga', 'unc', 'usc', 'utn' ],
		'Great Lake'  => [ 'uil', 'uin', 'uky', 'umi', 'uoh', 'upa' ],
		'Southwest'   => [ 'uaz', 'uca', 'uco', 'unv', 'unm', 'uut' ],
		'Northwest'   => [ 'uid', 'umt', 'une', 'uor', 'uwa', 'uwy' ],
		'Midwest'     => [ 'uia', 'uks', 'umn', 'und', 'usd', 'uwi' ],
		'South'       => [ 'uar', 'ula', 'ums', 'umo', 'uok', 'utx' ] );


sub get_roster {

    my ($stats_filename) = $_[0].".stat";
    my ($input);

    open( $input, "$stats_program $stats_filename|");

    while (<$input>) {

	print "$_";
    }

    close( $input );
}

sub get_record {

    my ($stats_filename) = $_[0].".stat";
    my (@details);

    if ( -e $stats_filename ) {

	@details = split( ",", `$record_program $stats_filename` );
    }
    else {

	@details = ( $_[0], 0, 0, 0, 0, 0 );
    }

    return \@details;
}

foreach $conference (@conferences) {

    print "$conference\n";
    print "\n";

    foreach $team ( @{ $teams{$conference} } ) {

	my ($record) = get_record( $team );

	printf "%s  %2d - %2d - %2d\n", $record->[0], $record->[1], $record->[2], $record->[3];
	print "\n";

	get_roster( $team );

	print "\n";
    }
}

exit;
